test/common/http/hcm_router_fuzz_test.cc

```diff
@@ -538,6 +538,10 @@ class Harness {
       }
       case ActionCase::kRequestHeader: {
         auto& a = action.request_header();
+        if (a.has_headers() || a.headers().headers_size() == 0) {
```

Comment:
Not sure I fully grok, why is this breaking if `a.has_headers()` evaluates to `true`?
Taking a step back, is the need here to have headers() defined? If so, would a PGV `[(validate.rules).message.required = true]` make more sense (see [example](https://github.com/envoyproxy/envoy/blob/c1dcfad275232750c8df15f693da5deae34d1792/test/common/http/codec_impl_fuzz.proto#L16))?
Furthermore, should the sanitized headers ([Fuzz::fromHeaders<TestRequestHeaderMapImpl>](https://github.com/envoyproxy/envoy/blob/c1dcfad275232750c8df15f693da5deae34d1792/test/fuzz/utility.h#L86)) be used in this case?

Reply:
Oh, that's a typo. It should be !a.has_headers(). But anyway, using PGV is a better idea.  The other thing is for input
actions {
  request_header {
   headers {
  }
  }
}
actions {
  request_header {
   headers {
  }
  }
}

adding a PGV here: https://github.com/envoyproxy/envoy/blob/c1dcfad275232750c8df15f693da5deae34d1792/test/fuzz/common.proto#L16 to make the header list size >=1 will work. But this PGV impacts a lot of other components. So, just adding an if check here to achieve the same goal.